src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTrace.netcore50aot.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Internal.Diagnostics;
+
+namespace System.Diagnostics
+{
+    public sealed class StackTrace
+    {
+        private readonly IntPtr[] _ips;
+        private readonly bool _fNeedFileInfo;
+
+        public StackTrace(Exception exception, bool needFileInfo)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException("exception");
+            }
+            _ips = ExceptionExtensions.GetStackIPs(exception);
+            _fNeedFileInfo = needFileInfo;
+        }
+
+        public StackFrame[] GetFrames()
+        {
+            // There are special IPs mixed in to the list that aren't intended to be exposed as StackFrames
+            // Deliberately doing this without Linq to keep the dependency set down.
+            StackFrame[] tempFrames = new StackFrame[_ips.Length];
+            int countNonSpecialIPs = 0;
+            for (int i = 0; i < _ips.Length; i++)
```

Comment:
It may be beneficial to copy `_ips` into a local variable for potential improved JIT code gen.